core/config/v2/types.go

```diff
@@ -165,34 +165,124 @@ func validateDBURL(dbURI url.URL) error {
 }
 
 func (d *DatabaseSecrets) ValidateConfig() (err error) {
-	if d.AllowSimplePasswords && build.IsProd() {
+	if *d.AllowSimplePasswords && build.IsProd() {
```

Comment:
Does these need to be nil checked now?

Reply:
What would be idiomatic? My desired behavior would be to panic - I want to enforce any code path leading to this method sets`AllowSimplePasswords` beforehand.

Reply:
That's fine as long as it's accounted for :shrug: 
I just wasn't sure since we don't have a non-nil invariant by default for secrets like with config.